pkg/api/customization/globalresource/globalnamespace_store.go

```diff
@@ -0,0 +1,50 @@
+package globalresource
+
+import (
+	"github.com/rancher/norman/types"
+	"github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/client/management/v3"
+	"github.com/sirupsen/logrus"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"fmt"
+	"strings"
+)
+
+const (
+	cattleGlobalNamespace = "cattle-global-data"
+)
+
+//this store makes sure that the global resources are assigned to a global namespace, it creates one if not already present.
+type GlobalNamespaceStore struct {
+	types.Store
+	NamespaceInterface v1.NamespaceInterface
+}
+
+func (s *GlobalNamespaceStore) Create(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}) (map[string]interface{}, error) {
+	logrus.Infof("Creating a Global CR, assigning to %v namespace", cattleGlobalNamespace)
```

Comment:
Nit: I think we should:

* be more explicit in logging the type of the object, CR is too generic
* we should log it in debug

Reply:
done